sdk/storage/AzureStorageBlob/Source/Options/ClientOptions.swift

```diff
@@ -33,34 +33,57 @@ public struct StorageBlobClientOptions: AzureConfigurable {
     public let apiVersion: String
     /// The `ClientLogger` to be used by this `StorageBlobClient`.
     public let logger: ClientLogger
-
-    internal let downloadNetworkPolicy: TransferNetworkPolicy
-
-    internal let uploadNetworkPolicy: TransferNetworkPolicy
+    /// Options for configuring telemetry sent by this `StorageBlobClient`.
+    public let telemetryOptions: TelemetryOptions
 
     // Blob operations
 
+    /// An identifier used to associate this client with transfers it creates. When a transfer is reloaded from disk
+    /// (e.g. after an application crash), it can only be resumed once a client with the same `restorationId` has been
+    /// initialized. If your application only uses a single `StorageBlobClient`, it is recommended to use a value unique
+    /// to your application (e.g. "MyApplication"). If your application uses multiple clients with different
+    /// configurations, use a value unique to both your application and the configuration (e.g.
+    /// "MyApplication.userClient").
+    public let restorationId: String
+
     /// The maximum size of a single chunk in a blob upload or download.
     public let maxChunkSizeInBytes: Int
 
+    /// The `TransferNetworkPolicy` to use for managed downloads.
+    public let downloadNetworkPolicy: TransferNetworkPolicy
+
+    /// The `TransferNetworkPolicy` to use for managed uploads.
+    public let uploadNetworkPolicy: TransferNetworkPolicy
```

Comment:
Why are these public instead of internal?

Reply:
Since the customer passes them in to the constructor of the `Options` object, it seems like they should be able to inspect them from the object. The `Options` objects are really just structs that hold onto a bunch of user-supplied options and contain no logic, so I feel like each option should be:

1. Provided to the initializer by the customer.
2. Have a default value in the initializer so it can be optional.
3. Be reflected in a public immutable property inside the struct.

I didn't see a reason why the policies should be an exception to this.